src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -965,6 +967,29 @@ public void setFilter(String filterName) {
         this.filterName = filterName;
     }
 
+    /**
+     * Get replicator filter. If the replicator is a pull replicator,
+     * calling this method will return null.
+     *
+     * This is equivalent to CBL_ReplicatorSettings's compilePushFilterForDatabase:status:.
+     */
+    public ReplicationFilter compilePushReplicationFilter() {
```

Comment:
No because it is used in Replication.java as well. We should adopt ReplicationSettings from iOS.

https://github.com/couchbase/couchbase-lite-java-core/issues/870